binary_io/src/lib.rs

```diff
@@ -1,24 +1,42 @@
-/*! Encoding/decoding traits
-*/
-
 pub use nom::IResult;
 pub use cookie_factory::GenError;
 
 use nom::number::streaming::{le_u8, le_u16};
+use nom::{named, map, count};
+use cookie_factory::{do_gen, gen_be_u8, gen_le_u16};
+
+#[cfg(feature = "sodiumoxide")]
```

Comment:
What's the purpose of this feature? You use it in almost every line, so it might be better to move all this to a separate module and define it with this feature.

Reply:
Orphan rules. You can't implement a foreign trait for a foreign type.

Reply:
Aren't they per crate?

Reply:
Indeed.